baybe/objectives/deprecation.py

```diff
@@ -0,0 +1,59 @@
+"""Temporary functionality for backward compatibility."""
+
+from __future__ import annotations
+
+import warnings
+from typing import TYPE_CHECKING, Union
+
+from cattrs.gen import make_dict_structure_fn
+
+from baybe.serialization import converter
+from baybe.targets.base import Target
+from baybe.utils.basic import get_subclasses
+
+if TYPE_CHECKING:
+    from baybe.objectives.base import Objective
+
+
+def structure_objective(val: dict, _) -> Objective:
+    """A structure hook that automatically determines an objective fallback type."""  # noqa: D401 (imperative mood)
+    from baybe.objective import Objective as factory
```

Comment:
dont you think `OldObjective` or `DeprecatedObjective` is clearer? of course these class obejcts are factories, but im that name makes it much clearer later on